### PR TITLE
BIOWDL-371: Update TALON output to align with new version.

### PR DESCRIPTION
**Pull request by JasperBoom**
### Checklist
- [x] Pull request details were added to CHANGELOG.md

### CODE REVIEW

File: talon.wdl [hunk 1]
```diff
@@ -446,39 +446,38 @@ task SummarizeDatasets {
 
 task Talon {
     input {
-        File SAMfile
```
Comment (DavyCats) [inline talon.wdl]: SAM files are specified in the config files right? How are they getting localized now?

Side question, can we use BAM files instead?

Comment (JasperBoom) [reply talon.wdl]: Correct, they are specified in the config file. I now generate the config file in the main TALON-WDL pipeline. It uses the SAM files in the specified output directory.

Comment (JasperBoom) [reply talon.wdl]: Sadly BAM has not yet been implemented. pysam is now being used for the parsing of the SAM file, but the BAM option is not yet provided, we might have to implement that ourselves at some point.

Comment (DavyCats) [reply talon.wdl]: SAM files still need to get localized to the task, they should still be provided as an input.

File: talon.wdl [hunk 2]
```diff
@@ -446,39 +446,38 @@ task SummarizeDatasets {
...
         Int cores = 4
-        String memory = "20G"
+        String memory = "25G"
         String dockerImage = "biocontainers/talon:v4.4.1_cv1"
     }
 
     command {
         set -e
         mkdir -p $(dirname ~{outputPrefix})
         mv ${configFile} ./${configFileName}
```
Comment (DavyCats) [inline talon.wdl]: Why is this `mv` needed?

Comment (JasperBoom) [reply talon.wdl]: Ah yes thank you, that was a left-over that should now be fixed by export TMPDIR=/tmp.

File: talon.wdl (hunk 1 above)
Comment (JasperBoom) [reply talon.wdl]: But they wouldn't be the same SAM files as in the config file in that case, so TALON wouldn't use them. Unless I alter or create the config file in the talon task. Would that be preferable to creating it in a separate task in the pipeline.wdl of TALON-WDL?

Comment (DavyCats) [reply talon.wdl]: Better to make the config in the same task. Right now, this task will only work if a custom configuration is used, which mounts whatever directory the SAM files are in.

Comment (JasperBoom) [reply talon.wdl]: TALON wants the absolute path to the SAM files, or it has to be in the same folder as the TALON process is running from (I think thats how I used to make it work with the mv's, but it was for single SAM file runs).

Comment (DavyCats) [reply talon.wdl]: I believe cromwell will make them absolute paths if they are of type `File`

Comment (JasperBoom) [reply talon.wdl]: I have now included some bash code to create the config file.

File: talon.wdl [hunk 3]
```diff
@@ -446,39 +446,43 @@ task SummarizeDatasets {
...
-        mv ${SAMfile} ./${SAMfileName}
+        export TMPDIR=/tmp
+        for file in ~{sep=" " SAMfiles}
+        do
+            configFileLine="$(basename ${file%.*}),~{organism},~{sequencingPlatform},${file}"
+            echo ${configFileLine} >> ~{outputPrefix}/talonConfigFile.csv
+        done
         talon \
-        ~{"--f " + configFileName} \
+        --f + talonConfigFile.csv
```
Comment (DavyCats) [inline talon.wdl]:
```suggestion
        --f talonConfigFile.csv
```

**Pull request merged**